transport/rif_comms/node.py

```diff
@@ -176,23 +177,21 @@ def stop(self):
             if message_queue.greenlet:
                 message_queue.notify()  # if we need to send something, this is the time
 
-        if self.greenlet:
-            self.greenlet.kill()
-            self.greenlet.get()
```

Comment:
i am not sure it is correct to delete this.

what if `_run` exits through `except Exception`?

in that case `self.greenlet` will not have been stopped, therefore these original statements kind of make sense to me.

what do you think? could you please explain a bit more about this "incorrect shutdown behavior"?

Reply:
of run exits through except Exception it will call the stop method and lumino will be shutted down

i dont found any issues with self.greenlet being unstopped. Why? because that greenlet is just for the _run execution. The run execution was stopped by the exception, then the greenlet isnt running anyway, and also, the lumino process will be shutted down.
 
About the incorrect shutdown behavior, code hangs on greenlet.kill, dont know why, but i dont found coherent to kill a greenlet when a exception on the run method occurs (because the run method is the greenlet execution itself)

that code was "copied" from the matrix client implemention.

the matrix transport node stop method calls self._client.stop_listener_thread() and stop the client thread. I think thats because the greenlet of the matrix node transport is used for different things. It was used for the global msg worker, and the client was responsible of the sync with the matrix homeserver

since we handle the streams for comms message reception on the transport node (not in the rif comms client), the behaviour is different